src/app/modules/item/services/item-task-answer.service.ts

```diff
@@ -121,4 +114,26 @@ export class ItemTaskAnswerService implements OnDestroy {
   clearAnswer(): Observable<unknown> {
     return this.task$.pipe(take(1), switchMap(task => task.reloadAnswer('')));
   }
+
+  private saveAnswerAndStateInterval(savedAnswer: string, savedState: string): Observable<{ success: boolean }> {
+    return interval(answerAndStateSaveInterval).pipe(
+      switchMap(() => this.task$),
+      switchMap(task => forkJoin({ answer: task.getAnswer(), state: task.getState() })),
+      startWith({ answer: savedAnswer, state: savedState }), // start with saved answer & state to keep only changed answer & state
+      distinctUntilChanged((a, b) => a.answer === b.answer && a.state === b.state),
+      skip(1), // skip currently saved answer & state tuple ...
+      take(1), // ... and take next unsaved one
+      withLatestFrom(this.config$),
+      switchMap(([{ answer, state }, { route, attemptId }]) =>
+        this.currentAnswerService.update(route.id, attemptId, { answer, state }).pipe(
+          mapTo({ success: true }),
+          catchError(() => of({ success: false })),
+          switchMap(result => concat(
+            of(result), // emit success state
+            this.saveAnswerAndStateInterval(answer, state)), // restart treatment
```

Comment:
I cannot convince myself that it is a good thing to do this in Rx, but I acknowledge it is not an easy case.
Wasn't it achievable using `repeatWhen`?

Reply:
I tested behavior of `repeat` operator (using [rxviz](https://rxviz.com/)), it appears to be exactly what we need here :ok_hand: 
I'll also retest the app

Reply:
It works, the only problem I had was :
I can't repeat as is the observable because I need to take last _saved_ values at each repeat. I opted for an internal thunk `saved` holding the saved values, and I use this object to filter from the task only unsaved values

It looks more rx, but I find not completely satisfying. I had other methods in mind like using a Su bject, but technically it's same result: when receiving a result, populate a variable external to the observable with saved values and re-read it in observable next value to apply a filter operation